app/uk/gov/hmrc/cdsreimbursementclaimfrontend/models/reimbursement/ReimbursementState.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cdsreimbursementclaimfrontend.models.reimbursement
+
+final case class ReimbursementState(
+  dutyTypesAnswer: DutyTypesAnswer,
+  dutyCodesAnswer: DutyCodesAnswer
+)
+
+object ReimbursementState {
+
+  def computeReimbursementState(
+    dutyTypesAnswer: DutyTypesAnswer,
+    dutyCodesAnswer: DutyCodesAnswer
+  ): ReimbursementState = {
+
+    val dutyCodesForCurrentlySelectedDutyTypes =
+      dutyCodesAnswer.dutyCodes
+        .filter(dutyTypesToDutyCodesMap =>
+          dutyTypesAnswer.dutyTypesSelected
+            .contains(dutyTypesToDutyCodesMap._1)
+        )
+
+    val dutyCodesForNewlySelectedDutyTypes =
+      dutyTypesAnswer.dutyTypesSelected
+        .filterNot(dutyType =>
```

Comment:
I reckon filterNot can be replaced with diff
`dutyTypesAnswer.dutyTypesSelected.diff(dutyCodesForCurrentlySelectedDutyTypes.keys.toList)`

Reply:
Yeah, it can be. I had a previous implementation where I converted them to `Sets`. If I use combinators like diff and intersect I have a mental model of a `Set` which I don't with a `Map`